packages/apps/human-app/server/src/integrations/external-api/external-api.gateway.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from '@nestjs/common';
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+import { lastValueFrom } from 'rxjs';
+import {
+  UserStatisticsCommand,
+  UserStatisticsResponse,
+} from '../../modules/statistics/interfaces/user-statistics.interface';
+import { HttpService } from '@nestjs/axios';
+import {
+  OracleStatisticsCommand,
+  OracleStatisticsResponse,
+} from '../../modules/statistics/interfaces/oracle-statistics.interface';
+import {
+  JobAssignmentCommand,
+  JobAssignmentData,
+  JobAssignmentParams,
+  JobAssignmentResponse,
+  JobsFetchParams,
+  JobsFetchParamsCommand,
+  JobsFetchParamsData,
+  JobsFetchResponse,
+} from '../../modules/job-assignment/interfaces/job-assignment.interface';
+import {
+  JobsDiscoveryParams,
+  JobsDiscoveryParamsCommand, JobsDiscoveryParamsData,
+  JobsDiscoveryResponse,
+} from '../../modules/jobs-discovery/interfaces/jobs-discovery.interface';
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+
+@Injectable()
+export class ExternalApiGateway {
+  constructor(
+    private httpService: HttpService,
+    @InjectMapper() private mapper: Mapper,
+  ) {}
+  private async callExternalHttpUtilRequest<T>(
+    options: AxiosRequestConfig,
+  ): Promise<T> {
+    const response: AxiosResponse<T> = await lastValueFrom(
+      this.httpService.request(options),
+    );
+    return response.data;
+  }
+  async fetchUserStatistics(
+    command: UserStatisticsCommand,
+  ): Promise<UserStatisticsResponse> {
+    const options: AxiosRequestConfig = {
+      method: 'GET',
+      url: `${command.oracleUrl}/stats/assignment`,
+      headers: {
+        Authorization: `Bearer ${command.token}`,
+      },
+    };
+    return this.callExternalHttpUtilRequest<UserStatisticsResponse>(options);
+  }
+  async fetchOracleStatistics(
+    command: OracleStatisticsCommand,
+  ): Promise<OracleStatisticsResponse> {
+    const options: AxiosRequestConfig = {
```

Comment:
I thought about extracting creating options as a separate function but decided against this as this MR is already pretty big and there is already a lot to test in it without adding this - something to consider in the future